cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/Group.java

```diff
@@ -41,69 +43,71 @@ public class Group extends DynamicModel {
-            rootVolumeSize, identity, deletedInstances, groupNetwork, tags);
+                rootVolumeSize, identity, deletedInstances, groupNetwork, tags);
     }
 
     public Group(String name,
-        InstanceGroupType type,
-        Collection<CloudInstance> instances,
-        Security security,
-        CloudInstance skeleton,
-        Map<String, Object> parameters,
-        InstanceAuthentication instanceAuthentication,
-        String loginUserName,
-        String publicKey,
-        int rootVolumeSize,
-        Optional<CloudFileSystemView> identity,
-        GroupNetwork groupNetwork,
-        Map<String, String> tags) {
+            InstanceGroupType type,
+            Collection<CloudInstance> instances,
+            Security security,
+            CloudInstance skeleton,
+            Map<String, Object> parameters,
+            InstanceAuthentication instanceAuthentication,
+            String loginUserName,
+            String publicKey,
+            int rootVolumeSize,
+            Optional<CloudFileSystemView> identity,
+            GroupNetwork groupNetwork,
+            Map<String, String> tags) {
         this(name, type, instances, security, skeleton, parameters, instanceAuthentication, loginUserName, publicKey,
-            rootVolumeSize, identity, new ArrayList<>(), groupNetwork, tags);
-    }
-
-    public Group(String name,
-        InstanceGroupType type,
-        Collection<CloudInstance> instances,
-        Security security,
-        CloudInstance skeleton,
-        Map<String, Object> parameters,
-        InstanceAuthentication instanceAuthentication,
-        String loginUserName,
-        String publicKey,
-        int rootVolumeSize,
-        Optional<CloudFileSystemView> identity,
-        List<CloudInstance> deletedInstances,
-        GroupNetwork groupNetwork,
-        Map<String, String> tags) {
+                rootVolumeSize, identity, new ArrayList<>(), groupNetwork, tags);
+    }
+
+    @JsonCreator
+    public Group(
+            @JsonProperty("name") String name,
+            @JsonProperty("type") InstanceGroupType type,
+            @JsonProperty("instances") Collection<CloudInstance> instances,
+            @JsonProperty("security") Security security,
+            @JsonProperty("skeleton") CloudInstance skeleton,
+            @JsonProperty("parameters") Map<String, Object> parameters,
+            @JsonProperty("instanceAuthentication") InstanceAuthentication instanceAuthentication,
+            @JsonProperty("loginUserName") String loginUserName,
+            @JsonProperty("publicKey") String publicKey,
+            @JsonProperty("rootVolumeSize") int rootVolumeSize,
+            @JsonProperty("identity") Optional<CloudFileSystemView> identity,
+            @JsonProperty("deletedInstances") List<CloudInstance> deletedInstances,
+            @JsonProperty("network") GroupNetwork groupNetwork,
```

Comment:
same here, the value of jsonproperty became different than name of parameter, is this backward compatible?

Reply:
it is intentional as the field name is `network`, its getter name is `public GroupNetwork getNetwork()` and therefore the Json Property name must be `network` too